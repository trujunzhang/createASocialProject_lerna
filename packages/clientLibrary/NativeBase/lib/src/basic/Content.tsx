import * as React from 'react'
import * as ReactNative from 'react-native'
import * as PropTypes from 'prop-types'
import { KeyboardAwareScrollView } from '@app/react-native-keyboard-aware-scroll-view'
import { connectStyle } from '@app/native-base-shoutem-theme'
import mapPropsToStyleNames from '../utils/mapPropsToStyleNames'
import { platformVariables as variable } from '@app/native-base-variables'
import { isIphoneX } from '@app/react-native-iphone-x-helper'

export interface IContentProps {
  style?: ReactNative.ViewStyle | Array<ReactNative.ViewStyle> | any
  /**
     * The theme prop can be applied to any component of NativeBase.
     */
  refreshing?: boolean
  refreshControl?: object | any
  theme?: Object
  padder?: boolean
  disableKBDismissScroll?: boolean | any
  enableResetScrollToCoords?: boolean
  contentOffset?: Object | any
  scrollEnabled?: boolean
  contentContainerStyle?: ReactNative.ViewStyle | Array<ReactNative.ViewStyle>
  keyboardShouldPersistTaps?: string | any
  keyboardDismissMode?: string | any
}
class Content extends React.Component<IContentProps, any> {
  private _root: any
  private _scrollview: any
  static contextTypes = {
    theme: PropTypes.object
  }
  constructor(props) {
    super(props)
    this.state = {
      orientation: 'portrait'
    }
  }
  layoutChange(val) {
    let maxComp = Math.max(variable.deviceWidth, variable.deviceHeight)
    if (val.width >= maxComp) this.setState({ orientation: 'landscape' })
    else {
      this.setState({ orientation: 'portrait' })
    }
  }

  calculateLeft(mode, inSet) {
    let inset: any = null
    if (inSet !== undefined) {
      inset = inSet
    } else {
      inset = variable.Inset
    }
    const InsetValues = mode === 'portrait' ? inset.portrait : inset.landscape
    let leftPadder = null
    if (this.props.style[1] !== undefined) {
      if (
        this.props.style[1].padding !== undefined ||
        this.props.style[1].paddingLeft !== undefined
      ) {
        leftPadder =
          (this.props.style[1].paddingLeft
            ? this.props.style[1].paddingLeft
            : this.props.style[1].padding) + InsetValues.leftInset
      }
    } else if (
      this.props.style.padding !== undefined ||
      this.props.style.paddingLeft !== undefined
    ) {
      leftPadder =
        (this.props.style.paddingLeft ? this.props.style.paddingLeft : this.props.style.padding) +
        InsetValues.leftInset
    } else {
      leftPadder = InsetValues.leftInset
    }
    return leftPadder
  }
  calculateRight(mode, inSet) {
    let inset: any = null
    if (inSet !== undefined) {
      inset = inSet
    } else {
      inset = variable.Inset
    }
    const InsetValues = mode === 'portrait' ? inset.portrait : inset.landscape
    let rightPadder = null
    if (this.props.style[1] !== undefined) {
      if (
        this.props.style[1].padding !== undefined ||
        this.props.style[1].paddingRight !== undefined
      ) {
        rightPadder =
          (this.props.style[1].paddingRight
            ? this.props.style[1].paddingRight
            : this.props.style[1].padding) + InsetValues.rightInset
      }
    } else if (
      this.props.style.padding !== undefined ||
      this.props.style.paddingRight !== undefined
    ) {
      rightPadder =
        (this.props.style.paddingRight ? this.props.style.paddingRight : this.props.style.padding) +
        InsetValues.rightInset
    } else {
      rightPadder = InsetValues.rightInset
    }
    return rightPadder
  }
  render() {
    const variables = this.context.theme
      ? this.context.theme['@@shoutem.theme/themeStyle'].variables
      : variable
    return isIphoneX() ? (
      <KeyboardAwareScrollView
        automaticallyAdjustContentInsets={false}
        resetScrollToCoords={(this.props.disableKBDismissScroll ? null : { x: 0, y: 0 }) as any}
        keyboardShouldPersistTaps={
          this.props.keyboardShouldPersistTaps ? this.props.keyboardShouldPersistTaps : 'handled'
        }
        ref={(c) => {
          this._scrollview = c
          this._root = c
        }}
        {...this.props}
        onLayout={(e) => this.layoutChange(e.nativeEvent.layout)}
        style={[
          this.props.style,
          {
            paddingLeft: this.calculateLeft(this.state.orientation, variables.Inset)
          }
        ]}
        contentContainerStyle={[
          { padding: this.props.padder ? variables.contentPadding : undefined },
          this.props.contentContainerStyle
        ]}>
        {this.props.children}
      </KeyboardAwareScrollView>
    ) : (
        <KeyboardAwareScrollView
          automaticallyAdjustContentInsets={false}
          resetScrollToCoords={(this.props.disableKBDismissScroll ? null : { x: 0, y: 0 }) as any}
          keyboardShouldPersistTaps={
            this.props.keyboardShouldPersistTaps ? this.props.keyboardShouldPersistTaps : 'handled'
          }
          ref={(c) => {
            this._scrollview = c
            this._root = c
          }}
          {...this.props}
          contentContainerStyle={[
            { padding: this.props.padder ? variables.contentPadding : undefined },
            this.props.contentContainerStyle
          ]}>
          {this.props.children}
        </KeyboardAwareScrollView>
      )
  }
}

// Content.propTypes = {
//     style: PropTypes.oneOfType([
//         PropTypes.object,
//         PropTypes.number,
//         PropTypes.array
//     ]),
//     padder: PropTypes.bool,
//     disableKBDismissScroll: PropTypes.bool,
//     enableResetScrollToCoords: PropTypes.bool,
//     keyboardShouldPersistTaps: PropTypes.string
// };

const StyledContent = connectStyle('NativeBase.Content', {}, mapPropsToStyleNames)(Content)

export { StyledContent as Content }
