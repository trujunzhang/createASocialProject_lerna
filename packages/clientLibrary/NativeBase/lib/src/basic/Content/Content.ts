import * as React from 'react'
import { ReactNative, RnTextStyleProp, RnViewStyleProp } from '@appUtils/rnCommonProps'

/**
 * see Widget Content.js
 */
export interface ContentProps {
  /**
   * The theme prop can be applied to any component of NativeBase.
   */
  refreshing?: boolean
  refreshControl?: object
  theme?: Object
  padder?: boolean
  disableKBDismissScroll?: boolean
  enableResetScrollToCoords?: boolean
  contentOffset?: Object
  scrollEnabled?: boolean
  style?: RnViewStyleProp | Array<RnViewStyleProp>
  contentContainerStyle?: RnViewStyleProp | Array<RnViewStyleProp>
  keyboardShouldPersistTaps?: string
  keyboardDismissMode?: string
}
