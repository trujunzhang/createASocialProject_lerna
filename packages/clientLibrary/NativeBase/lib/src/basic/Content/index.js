import * as React from 'react';
import * as PropTypes from 'prop-types';
// TODO:DJZHANG(NB)
import { getKeyboardAwareScrollView } from '@appUtils/cacheHelper';
import { isIphoneX } from 'react-primitives';
// import { KeyboardAwareScrollView } from 'react-native-keyboard-aware-scroll-view'
import { connectStyle } from 'native-base-shoutem-theme';
import { platformVariables as variable } from '@appVariables/theme/variables/platform'; // variables // [variable]
import { mapPropsToStyleNames } from '@app/tools';
class Content extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            orientation: 'portrait'
        };
    }
    layoutChange(val) {
        let maxComp = Math.max(variable.deviceWidth, variable.deviceHeight);
        if (val.width >= maxComp)
            this.setState({ orientation: 'landscape' });
        else {
            this.setState({ orientation: 'portrait' });
        }
    }
    calculateLeft(mode, inSet) {
        let inset = null;
        if (inSet !== undefined) {
            inset = inSet;
        }
        else {
            inset = variable.Inset;
        }
        const InsetValues = mode === 'portrait' ? inset.portrait : inset.landscape;
        let leftPadder = null;
        if (this.props.style[1] !== undefined) {
            if (this.props.style[1].padding !== undefined ||
                this.props.style[1].paddingLeft !== undefined) {
                leftPadder =
                    (this.props.style[1].paddingLeft
                        ? this.props.style[1].paddingLeft
                        : this.props.style[1].padding) + InsetValues.leftInset;
            }
        }
        else if (this.props.style.padding !== undefined ||
            this.props.style.paddingLeft !== undefined) {
            leftPadder =
                (this.props.style.paddingLeft ? this.props.style.paddingLeft : this.props.style.padding) +
                    InsetValues.leftInset;
        }
        else {
            leftPadder = InsetValues.leftInset;
        }
        return leftPadder;
    }
    calculateRight(mode, inSet) {
        let inset = null;
        if (inSet !== undefined) {
            inset = inSet;
        }
        else {
            inset = variable.Inset;
        }
        const InsetValues = mode === 'portrait' ? inset.portrait : inset.landscape;
        let rightPadder = null;
        if (this.props.style[1] !== undefined) {
            if (this.props.style[1].padding !== undefined ||
                this.props.style[1].paddingRight !== undefined) {
                rightPadder =
                    (this.props.style[1].paddingRight
                        ? this.props.style[1].paddingRight
                        : this.props.style[1].padding) + InsetValues.rightInset;
            }
        }
        else if (this.props.style.padding !== undefined ||
            this.props.style.paddingRight !== undefined) {
            rightPadder =
                (this.props.style.paddingRight ? this.props.style.paddingRight : this.props.style.padding) +
                    InsetValues.rightInset;
        }
        else {
            rightPadder = InsetValues.rightInset;
        }
        return rightPadder;
    }
    render() {
        const variables = this.context.theme
            ? this.context.theme['@@shoutem.theme/themeStyle'].variables
            : variable;
        const KeyboardAwareScrollView = getKeyboardAwareScrollView();
        return isIphoneX() ? (React.createElement(KeyboardAwareScrollView, Object.assign({ name: 'keyboardAwareScrollView', automaticallyAdjustContentInsets: false, resetScrollToCoords: this.props.disableKBDismissScroll ? null : { x: 0, y: 0 }, keyboardShouldPersistTaps: this.props.keyboardShouldPersistTaps ? this.props.keyboardShouldPersistTaps : 'handled', ref: (c) => {
                this._scrollview = c;
                this._root = c;
            } }, this.props, { onLayout: (e) => this.layoutChange(e.nativeEvent.layout), style: [
                this.props.style,
                {
                    paddingLeft: this.calculateLeft(this.state.orientation, variables.Inset)
                }
            ], contentContainerStyle: [
                { padding: this.props.padder ? variables.contentPadding : undefined },
                this.props.contentContainerStyle
            ] }), this.props.children)) : (React.createElement(KeyboardAwareScrollView, Object.assign({ name: 'keyboardAwareScrollView', automaticallyAdjustContentInsets: false, resetScrollToCoords: this.props.disableKBDismissScroll ? null : { x: 0, y: 0 }, keyboardShouldPersistTaps: this.props.keyboardShouldPersistTaps ? this.props.keyboardShouldPersistTaps : 'handled', ref: (c) => {
                this._scrollview = c;
                this._root = c;
            } }, this.props, { contentContainerStyle: [
                { padding: this.props.padder ? variables.contentPadding : undefined },
                this.props.contentContainerStyle
            ] }), this.props.children));
    }
}
Content.contextTypes = {
    theme: PropTypes.object
};
// Content.propTypes = {
//   style: PropTypes.oneOfType([PropTypes.object, PropTypes.number, PropTypes.array]),
//   padder: PropTypes.bool,
//   disableKBDismissScroll: PropTypes.bool,
//   enableResetScrollToCoords: PropTypes.bool,
//   keyboardShouldPersistTaps: PropTypes.string
// }
const StyledContent = connectStyle('NativeBase.Content', {}, mapPropsToStyleNames)(Content);
const TabContent = StyledContent;
export { StyledContent as Content, TabContent };
//# sourceMappingURL=index.js.map