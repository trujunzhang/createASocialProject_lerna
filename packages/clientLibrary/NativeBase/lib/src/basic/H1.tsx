import * as React from 'react'

import * as PropTypes from 'prop-types'
import { Text, TextProps } from 'react-primitives'

import { connectStyle } from '@app/native-base-shoutem-theme'
import mapPropsToStyleNames from '../utils/mapPropsToStyleNames'

export interface IH1Props extends TextProps {
  style?: ViewStyle | Array<ViewStyle>
}
class H1 extends React.Component<IH1Props, any> {
  private _root: any
  render() {
    return <Text ref={(c) => (this._root = c)} {...this.props} />
  }
}

const childrenType = function(props, propName, component) {
  let error
  const prop = props[propName]
  React.Children.forEach(prop, (child) => {
    if (typeof child !== 'string' && typeof child !== 'number') {
      error = new Error(`${component} should have only string or number`)
    }
  })
  return error
}

// H1.propTypes = {
// 	...Text.propTypes,
// 	children: childrenType,
// 	style: PropTypes.oneOfType([PropTypes.object, PropTypes.number, PropTypes.array]),
// };

const StyledH1 = connectStyle('NativeBase.H1', {}, mapPropsToStyleNames)(H1)

export { StyledH1 as H1 }
