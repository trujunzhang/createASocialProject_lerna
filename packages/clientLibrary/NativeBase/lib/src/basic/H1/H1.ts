import * as React from 'react'

import { ReactNative, RnViewStyleProp } from '@appUtils/rnCommonProps'

export interface H1Props extends ReactNative.TextProps {
  // Others
  center?: boolean
}
