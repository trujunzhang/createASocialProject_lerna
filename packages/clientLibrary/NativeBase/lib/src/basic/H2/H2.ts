import * as React from 'react'

import { ReactNative, RnViewStyleProp } from '@appUtils/rnCommonProps'

export interface H2Props extends ReactNative.TextProps {
  // Others
  center?: boolean
}
