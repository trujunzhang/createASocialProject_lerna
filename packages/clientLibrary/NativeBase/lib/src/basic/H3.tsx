import * as React from 'react'

import * as PropTypes from 'prop-types'
import { Text, TextProps ,ViewStyle} from 'react-primitives'

import { connectStyle } from '@app/native-base-shoutem-theme'
import mapPropsToStyleNames from '../utils/mapPropsToStyleNames'

export interface IH3Props extends TextProps {
  style?: ViewStyle | Array<ViewStyle>
}
class H3 extends React.Component<IH3Props, any> {
  private _root: any
  render() {
    return <Text ref={(c) => (this._root = c)} {...this.props} />
  }
}

const childrenType = function(props, propName, component) {
  let error
  const prop = props[propName]
  React.Children.forEach(prop, (child) => {
    if (typeof child !== 'string' && typeof child !== 'number') {
      error = new Error(`${component} should have only string or number`)
    }
  })
  return error
}

// H3.propTypes = {
// 	...Text.propTypes,
// 	children: childrenType,
// 	style: PropTypes.oneOfType([PropTypes.object, PropTypes.number, PropTypes.array]),
// };

const StyledH3 = connectStyle('NativeBase.H3', {}, mapPropsToStyleNames)(H3)

export { StyledH3 as H3 }
