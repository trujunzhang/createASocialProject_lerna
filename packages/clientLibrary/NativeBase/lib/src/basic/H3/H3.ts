import * as React from 'react'

import { ReactNative, RnViewStyleProp } from '@appUtils/rnCommonProps'

export interface H3Props extends ReactNative.TextProps {
  // Others
  center?: boolean
}
