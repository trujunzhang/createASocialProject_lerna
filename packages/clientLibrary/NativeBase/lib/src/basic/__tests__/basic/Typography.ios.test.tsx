import 'react-primitives'
import * as React from 'react'
import * as renderer from 'react-test-renderer'
import { Content } from '../../Content'
import { H1 } from '../../H1'
import { H2 } from '../../H2'
import { H3 } from '../../H3'
import { Text } from '../../Text'

// Note: test renderer must be required after react-native.

// jest.mock("Platform", () => {
//   const Platform = require.requireActual("Platform");
//   Platform.OS = "ios";
//  return Platform;
// });

it('renders Typography', () => {
  const tree = renderer
    .create(
      <Content>
        <H1>Header One</H1>
        <H2>Header Two</H2>
        <H3>Header Three</H3>
        <Text>Default</Text>
      </Content>
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})
