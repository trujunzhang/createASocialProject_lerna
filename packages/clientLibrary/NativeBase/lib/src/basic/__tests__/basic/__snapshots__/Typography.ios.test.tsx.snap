// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Typography 1`] = `
<view
  name="keyboardAwareScrollView"
  style={Object {}}
>
  <view
    name="View"
    style={Object {}}
  >
    <text
      style={Object {}}
    >
      Header One
    </text>
  </view>
  <view
    name="View"
    style={Object {}}
  >
    <text
      style={Object {}}
    >
      Header Two
    </text>
  </view>
  <view
    name="View"
    style={Object {}}
  >
    <text
      style={Object {}}
    >
      Header Three
    </text>
  </view>
  <view
    name="View"
    style={Object {}}
  >
    <text
      style={Object {}}
    >
      Default
    </text>
  </view>
</view>
`;
